Inc/HALALMock/Services/SharedMemory/SharedMemory.hpp

```diff
@@ -10,8 +10,15 @@ class SharedMemory {
 
     // do not use before SharedMemory::start, uninitialized!!!
     static EmulatedPin *gpio_memory;
+	static void* emulated_state_machine;
     static EmulatedPin &get_pin(Pin pin);
     static void start();
+	static void start_emulated_state_machine();
+	constexpr static size_t emulated_state_machine_size=16;
+	constexpr char* emulated_state_machine_name="/Shared_Memory_Emulated_State_Machine";
```

Comment:
Don't hardcode the name, what happens if we launch two instances of the simulator?

Reply:
Why would we launch to instances of the simulator, this shared memory should only be used for the state machine right? If not, should we generate an individual name for each time we launch an instance??

Reply:
two instances would be used for example if we wanted to run two boards in the same computer